packages/dev/core/src/Behaviors/Meshes/sixDofDragBehavior.ts

```diff
@@ -81,18 +81,36 @@ export class SixDofDragBehavior extends BaseSixDofDragBehavior {
         this._sceneRenderObserver = ownerNode.getScene().onBeforeRenderObservable.add(() => {
             if (this.currentDraggingPointerIds.length === 1 && this._moving && !this.disableMovement) {
                 // 1 pointer only drags mesh
-                const oldParent = ownerNode.parent;
-                ownerNode.setParent(null);
-                ownerNode.position.addInPlace(this._targetPosition.subtract(ownerNode.position).scale(this.dragDeltaRatio));
+                const deltaToAdd = TmpVectors.Vector3[0];
+                deltaToAdd.copyFrom(this._targetPosition).subtractInPlace(ownerNode.absolutePosition).scaleInPlace(this.dragDeltaRatio);
+                const deltaToAddTransformed = TmpVectors.Vector3[1];
+                deltaToAddTransformed.copyFrom(deltaToAdd);
+                // If the node has a parent, transform the delta to local space, so it can be added to the
+                // position in local space
+                if (ownerNode.parent) {
+                    const parentWorld = ownerNode.parent.getWorldMatrix();
+                    Vector3.TransformCoordinatesToRef(deltaToAdd, parentWorld, deltaToAddTransformed);

```

Comment:
If it is a delta, shouldn't it be `TransformNormalToRef` ?

Reply:
You're right! Thanks for catching that.